src/Raven.Server/Documents/PeriodicBackup/Restore/Sharding/SingleShardRestoreBackupTask.cs

```diff
@@ -27,42 +26,85 @@ internal class SingleShardRestoreBackupTask : RestoreBackupTask
 
             DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Aviv, DevelopmentHelper.Severity.Major, "test encrypted backup & restore");
             HasEncryptionKey = false;
+
+            _shardNumber = ShardHelper.GetShardNumber(DatabaseName);
+        }
+        
+        protected override Task OnBeforeRestoreAsync()
+        {
+            CreateDocumentDatabase();
             
-            _nonShardedDbName = ShardHelper.ToDatabaseName(DatabaseName);
+            var topology = RestoreSettings.DatabaseRecord.Sharding.Shards[_shardNumber];
+            Database.SetIds(topology, RestoreSettings.DatabaseRecord.Sharding.DatabaseId);
+
+            return Task.CompletedTask;
         }
 
-        protected override RavenConfiguration CreateDatabaseConfiguration()
+        protected override Task OnAfterRestoreAsync()
         {
-            return DatabasesLandlord.CreateDatabaseConfiguration(ServerStore, DatabaseName, RestoreSettings.DatabaseRecord.Settings);
+            SmugglerBase.EnsureProcessed(Result, skipped: false);
+            Progress.Invoke(Result.Progress);
+            return Task.CompletedTask;
         }
 
         protected override DatabaseRecord GetDatabaseRecord()
         {
             using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext serverContext))
             using (serverContext.OpenReadTransaction())
             {
-                var databaseRecord = ServerStore.Cluster.ReadDatabase(serverContext, _nonShardedDbName);
+                var databaseRecord = ServerStore.Cluster.ReadDatabase(serverContext, ShardHelper.ToDatabaseName(DatabaseName));
```

Comment:
why do we need to override `GetDatabaseRecord` ? can we have the same method and simply pass the database name?

Reply:
common implementation of `GetDatabaseRecord` is creating a new instance